### PR TITLE
Add event form

### PR DESCRIPTION
**Pull request by MohamedOmarii**
#13

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -12,6 +13,8 @@ server.set('views', path.join(__dirname, './', 'views'));
 
 server.use(express.static(path.join(__dirname, './', 'public')));
 server.use(router);
+server.use(bodyParser.urlencoded({ extended: true }));
+server.use(bodyParser.json());
```
Comment (mattlub) [inline server.js]: does the order of these `server.use(...` lines matter?

File: routes/add-event.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+const request = require('request');
+module.exports = (req, res) => {
+  var url = 'https://nazareth-open-tourism-platform.herokuapp.com/events';
+
+  request.post(url, (error, result) => {
```
Comment (mattlub) [inline routes/add-event.js]: you do not want to be doing `request.post` here. It is the handler for the add-event form page. for the `post` request, you should create another route.

File: routes/index.js [hunk 3]
```diff
@@ -2,8 +2,10 @@ const router = require('express').Router();
 
 const homePage = require('./home.js');
 const detailsPage = require('./event_details.js');
+const addEventPage = require('./add-event.js');
 
 router.get('/', homePage);
 router.get('/events/:id', detailsPage);
+router.get('/add-event', addEventPage);
```
Comment (mattlub) [inline routes/index.js]: add `router.post('/add-event, ...` here

Comment (MohamedOmarii) [reply routes/index.js]: :+1:

File: server.js (hunk 1 above)
Comment (MohamedOmarii) [reply server.js]: No, but for orderly

File: routes/add-event.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    startTime: req.body.startTime,
+    endTime: req.body.endTime,
+    cost: req.body.cost,
+    image: req.body.imageUrl,
+    en: {
+      name: req.body.name,
+      description: req.body.description
+    }
+  };
+  console.log(requestBody);
```
Comment (mattlub) [inline routes/add-event.js]: remove this!

File: routes/add-event.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
+const request = require('request');
+module.exports = (req, res) => {
+  var url = 'https://nazareth-open-tourism-platform.herokuapp.com/events';
+
+  console.log('form:', req.body);
```
Comment (mattlub) [inline routes/add-event.js]: remove!

File: routes/index.js [hunk 6]
```diff
@@ -2,8 +2,11 @@ const router = require('express').Router();
 
 const homePage = require('./home.js');
 const detailsPage = require('./event_details.js');
+const addEventPage = require('./add-event.js');
```
Comment (mattlub) [inline routes/index.js]: the name of this should be maybe `addEvent` instead of `addEventPage`

File: views/add-event.hbs [hunk 7]
```diff
@@ -0,0 +1,50 @@
+<div class="add-eventContainer">
+  <h3 class="userAddEvent">{user name} please add event</h3>
+
+  <form class="add-event-form" action="/add-event" method="post">
+    <label>Title: </label>
+    <input class="form-control" type="text" name="name" placeholder="title" required>
+
+    <label>Picture: </label>
+    <input class="form-control" type="file" name="imageUrl" accept="image/*" placeholder="choose picture">
```
Comment (mattlub) [inline views/add-event.hbs]: it doesn't work to upload an image like this.

Either:
- The image url should have type="text"
- or use a tool like S3 to handle image upload.

File: views/add-event.hbs [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+      <option value="cinema">Cinema</option>
+      <option value="sport">Sport</option>
+      <option value="cultural Event">Cultural</option>
+      <option value="guesthouse">Guesthouse</option>
+      <option value="store">Store</option>
+      <option value="restaurant">Restaurant</option>
+      <option value="historical site">Historical site</option>
+      <option value="hangout Spots">Hangout Spots</option>
+      <option value="miscellaneous">Miscellaneous</option>
+      <option value="unspecified">Unspecified</option>
```
Comment (mattlub) [inline views/add-event.hbs]: some of these options look like they are not really for `events`, but for `places`

Comment (mattlub) [reply views/add-event.hbs]: you should add your suggestions for event categories in the issue on the platform:
https://github.com/foundersandcoders/open-tourism-platform/issues/82

File: views/add-event.hbs [hunk 9]
```diff
@@ -0,0 +1,81 @@
+<div class="add-eventContainer">
+  <h3 class="userAddEvent">{user name} add event</h3>
+
+  <form class="add-event-form" action="/add-event" method="post">
+    <label>Title: </label>
+    <input class="form-control" type="text" name="name" placeholder="title" required>
+
+    <label>Picture: </label>
+    <input class="form-control" type="text" name="imageUrl" placeholder="choose picture">
+
```
Comment (mattlub) [inline views/add-event.hbs]: include a date field?

**Comment by mattlub** [comment]
you need to include a date field I think. This would also affect the js file, for sending the request.

also, the field for the image URL should have a better label.

File: routes/add-event.js [hunk 10]
```diff
@@ -0,0 +1,36 @@
+const request = require('request');
+module.exports = (req, res) => {
+  var url = 'https://nazareth-open-tourism-platform.herokuapp.com/events';
```
Comment (Karyum) [inline routes/add-event.js]: change this to const

File: views/add-event.hbs [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+    <label>Start Date:</label>
```
Comment (Karyum) [inline views/add-event.hbs]: add a placeholder for the date format

File: views/add-event.hbs [hunk 12]
```diff
@@ -0,0 +1,66 @@
...
+    <input class="form-control" type="date" name="startDate" required>
+
+    <label>End Date:</label>
```
Comment (Karyum) [inline views/add-event.hbs]: also a placeholder here

File: views/event_details.hbs [hunk 13]
```diff
@@ -15,7 +15,7 @@
 
   <div class="information">
     <h3  class="text-danger">categories: </h3> <h4 class="text-info">{{event.categories}}</h4>
-    <h3  class="text-danger">price: </h3> <h4 class="text-info">{{event.cost}}</h4>
+    <h3  class="text-danger">price: </h3> <h4 class="text-info">{{event.cost}} <img class="shekelIcon" src="../imges/shekel.png" alt="shekel icon"></h4>
```
Comment (mattlub) [inline views/event_details.hbs]: the src is wrong. You obviously didn't actually test this....
Also, the shekel icon shouldn't be in this PR, but I'll let it slide this time.

File: views/add-event.hbs [hunk 14]
```diff
@@ -0,0 +1,66 @@
...
+      <option value="competition">Competition</option>
+      <option value="launch">Launch</option>
+      <option value="party">Party</option>
+      <option value="wedding">Wedding</option>
+      <option value="cultural event">Cultural event</option>
+      <option value="Miscellaneous">Miscellaneous</option>
+    </select>
+
+    <label>Accessibility: </label>
+    <select class="accessibility form-control" name="accessibilityOptions" required>
```
Comment (mattlub) [inline views/add-event.hbs]: this should be a multi-select...

Comment (mattlub) [reply views/add-event.hbs]: so should categories. see https://github.com/foundersandcoders/open-tourism-platform/blob/master/docs.md#create-event

File: views/add-event.hbs [hunk 15]
```diff
@@ -0,0 +1,66 @@
...
+    <input class="form-control" type="text" name="name" placeholder="title" required>
+
+    <label>Picture: </label>
+    <input class="form-control" type="text" name="imageUrl" placeholder="choose picture">
+
+    <label>Start Date:</label>
+    <input class="form-control" type="date" name="startDate" placeholder="example year-month-day" required>
+
+    <label>End Date:</label>
+    <input class="form-control" type="date" name="endDate" placeholder="example year-month-day" required>
```
Comment (mattlub) [inline views/add-event.hbs]: I would say placeholder should be `yyyy-mm-dd`, this is more standard

File: views/add-event.hbs [hunk 16]
```diff
@@ -0,0 +1,66 @@
+<div class="add-eventContainer">
+  <h3 class="userAddEvent">{user name} add event</h3>
+
+  <form class="add-event-form" action="/add-event" method="post">
+    <label>Title: </label>
+    <input class="form-control" type="text" name="name" placeholder="title" required>
+
+    <label>Picture: </label>
+    <input class="form-control" type="text" name="imageUrl" placeholder="choose picture">
```
Comment (mattlub) [inline views/add-event.hbs]: the label should and placeholder be Image URL, or at least suggest it should be a URL

**Comment by mattlub** [comment]
screen shot 2017-09-11 at 11 42 44

screen shot 2017-09-11 at 11 44 07

- form shouldn't fill the whole page width
- { username } is written at the top
- there are errors in the console

**Comment by mattlub** [comment]
write test(s) for this route.

File: routes/add-event.js [hunk 17]
```diff
@@ -0,0 +1,36 @@
+const request = require('request');
+module.exports = (req, res) => {
+  const url = 'https://nazareth-open-tourism-platform.herokuapp.com/events';
+
+  const requestBody = {
+    categories: req.body.categories,
+    accessibilityOptions: req.body.accessibilityOptions,
+    startTime: req.body.startDate + 'T' + req.body.startTime,
+    endTime: req.body.endDate + 'T' + req.body.endTime,
```
Comment (mattlub) [inline routes/add-event.js]: add a timezone here?

File: views/event_details.hbs (hunk 13 above)
Comment (MohamedOmarii) [reply views/event_details.hbs]: I tested it, it was working but I change the name of the file , it was imges not images

**Pull request merged**